cheetah/particles/particle_beam.py

```diff
@@ -1033,6 +1033,56 @@ def linspaced(self, num_particles: int) -> "ParticleBeam":
             dtype=self.particles.dtype,
         )
 
+    def randomly_subsampled(
+        self,
+        num_particles: int,
+        adjust_particle_charges: bool = True,
+        random_state: torch.Generator | None = None,
+    ) -> "ParticleBeam":
+        """
+        Create a new beam with the same parameters as this beam, but with
+        `num_particles` particles randomly sampled from the original beam.
+
+        :param num_particles: Number of particles to create.
+        :param adjust_particle_charges: If True, the particle charges are adjusted
+            to match the total charge of the new beam.
```

Comment:
```suggestion
            to match the total charge of the old beam.
```

Maybe I misunderstood, but I guess the idea is to create new particles with larger charge such that the total charge of the old beam is preserved with fewer particles?